src/diaginterface/diaginterface.jl

```diff
@@ -88,3 +90,11 @@ function setdiagindices!(a::AbstractArray, v, i::Colon)
   diagview(a) .= v
   return a
 end
+
+"""
+    diagonal(v::AbstractVector) -> AbstractMatrix
+
+Return a diagonal matrix from a vector `v`. This is a replacement of `LinearAlgebra.Diagonal`
+that does not imply an output type. Defaults to `Diagonal(v)`.
+"""
+diagonal(A::AbstractVector) = LinearAlgebra.Diagonal(A)
```

Comment:
What do you think about also allowing axes to be passed, i.e.:
```julia
function diagonal(A::AbstractVector)
  axis = only(axes(A))
  return diagonal(A, (axis, axis))
end
diagonal(A::AbstractVector, ax) = LinearAlgebra.Diagonal(A)
```
That would allow extra generality to make rectangular diagonal arrays, higher dimensional diagonal arrays, add sector and dual information, etc.